### ISSUE TITLE
PassiveMpc _reconstruct hangs when future result is set multiple times

### ISSUE THREAD

**Issue by smkuls** [initc3/HoneyBadgerMPC#37]
When a party gets `t+1` shares we do `self._openings[shareid].set_result(s)`.
The present code:
```
if len(shares) < self.t+1:
            raise NotEnoughShares
        # print('[%d] reconstruct %s' % (self.myid, shareid,))
        s = Poly.interpolate_at(shares)
        # Set the result on the future representing this share
        self._openings[shareid].set_result(s)
```
Consider n=6, t=1, once we get two shares we set the result, but now when we receive the third share it still attempts to set the result which was previously set and it hangs (`set_result` as per the documentation throws an exception when the result has been set previously which for some reason is not getting displayed).

It is working in the present cases because we have n=3, t=2 (i.e. n == t+1)

**Comment by amiller** [comment]
Also we need to double check the degree of polynomials we plan to use. 
  If we are using `N=6` and `t=1`, it should be safe to use either a degree=1 polynomial or a degree=4 polynomial. The decision should be based on which is best for performance in batch operation. I believe @tyurek is looking into this...

**Comment by amiller** [comment]
We can't figure out why this doesn't throw an exception. Can we reproduce the basic problem of calling `fut.set_result()` twice?

**Comment by smkuls** [comment]
I created #39 for that. 

I think it is better to down tasks to separate issues/PRs to keep the PR shorter which makes it easier to review and also keeps the history cleaner. 

What do you think?

**Comment by amiller** [comment]
I think this is closed with #38
